### PR TITLE
Implement reset interface in the physics system

### PR DESCRIPTION
**Pull request by mjcarroll**
This builds upon the reset interface for systems introduced in #1249 by implementing the interface for the `Physics` system.

Upon receiving a reset command, this allows the physics plugin to rewind to the initial state of the world without removing and re-adding all of the entities.

### CODE REVIEW

File: test/integration/reset.cc [hunk 1]
```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (C) 2018 Open Source Robotics Foundation
```
Comment (azeey) [inline test/integration/reset.cc:2]:
```suggestion
 * Copyright (C) 2022 Open Source Robotics Foundation
```

**Review by azeey — APPROVED** [review]

File: examples/plugin/reset_plugin/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.10.2 FATAL_ERROR)
+
+find_package(ignition-cmake2 REQUIRED)
+
+project(ResetPlugins)
+find_package(ignition-gazebo7 REQUIRED)
+add_library(JointPositionRandomizer SHARED joint_position_randomizer.cc)
```
Comment (chapulina) [inline examples/plugin/reset_plugin/CMakeLists.txt:7]: Follow the pattern of `CamelCase` for plugin files

```suggestion
add_library(JointPositionRandomizer SHARED JointPositionRandomizer.cc)
```

File: examples/plugin/reset_plugin/joint_position_randomizer.cc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  void Reset(const UpdateInfo &_info, EntityComponentManager &_ecm) override
+  {
+    std::cout << "Reset joints\n";
+    auto joints = _ecm.EntitiesByComponents(
+        components::ParentEntity(this->targetEntity), components::Joint());
+    for (auto joint : joints)
+    {
+      double pos = math::Rand::DblUniform(0, IGN_PI);
```
Comment (chapulina) [inline examples/plugin/reset_plugin/joint_position_randomizer.cc:47]: We probably want a different range for the prismatic joint

resetplugin

**Review by chapulina — COMMENTED** [review]
It would be nice to add a full description for this PR. It only refers to another PR that doesn't explain anything about this one.

Are there plans to support models that are created or removed during the simulation? This limitation should be documented somewhere.

The PR looks good to me besides the comments below 👍🏽

File: examples/plugin/reset_plugin/joint_position_randomizer.cc (hunk 3 above)
Comment (mjcarroll) [reply examples/plugin/reset_plugin/joint_position_randomizer.cc]: Done in [3799ff3](https://github.com/ignitionrobotics/ign-gazebo/pull/1327/commits/3799ff3f5e4030e0f0351c56b05fa6a06295fd87)

File: examples/plugin/reset_plugin/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.10.2 FATAL_ERROR)
+
+find_package(ignition-cmake2 REQUIRED)
```
Comment (scpeters) [inline examples/plugin/reset_plugin/CMakeLists.txt:3]: cmake3

Comment (scpeters) [reply examples/plugin/reset_plugin/CMakeLists.txt:3]: actually, I'm not sure why this example needs to find `ignition-cmake` at all

**Pull request merged**